index.js

```diff
@@ -129,6 +134,25 @@ module.exports.init = function () {
       unix_sockets.push(address.address);
       fs.chmodSync(address.address, '664');
     }
+  }).on('exit', function (worker, code, signal) {
+    const pid = worker.process.pid;
+
+    if (worker._worker_terminated) {
+      // this is used here to distinguish expected/unexpected worker deaths.
+      if (code === 0) {
+        debug('PID/%s: terminated worker has exited', pid);
+      } else {
+        debug('PID/%s: terminated worker has crashed: %o', pid, { code, signal });
+      }
+    } else {
+      const uptime = Date.now() - worker._worker_started;
+      const restartDelay = Math.max(0, RESTART_DELAY - uptime);
+
+      debug('PID/%s: worker has crashed: %o', pid, { code, signal, uptime, restartDelay });
+      setTimeout(() => {
+        fork(worker._worker_index, reload_counter);
```

Comment:
Should we skip these actions on `fork` function when the worker already exited?
https://github.com/jfromaniello/master-process/blob/80875976039361618bff4d534bd9e88e2c4c15cf/index.js#L64-L73

Reply:
I think that whole logic needs to be improved as it has other problems, see https://github.com/jfromaniello/master-process/issues/10. I propose to do so in another PR.

In case of a worker crash the `reload_counter` is unchanged and so the actions are effectively skipped.